website/teacher.py

```diff
@@ -317,6 +319,97 @@ def remove_invite(user):
         DATABASE.remove_class_invite(username)
         return {}, 200
 
+    @app.route('/for-teachers/create-accounts', methods=['GET'])
+    @requires_login
+    def create_accounts(user):
+        if not is_teacher(user):
+            return 'Only teachers can create multiple accounts', 403
+        classes = DATABASE.get_teacher_classes(user['username'], False)
+
+        return render_template('create-accounts.html', classes=classes)
+
+    @app.route('/for-teachers/create-accounts', methods=['POST'])
+    @requires_login
+    def store_accounts(user):
+        if not is_teacher(user):
+            return 'Only teachers can create multiple accounts', 403
+        body = request.json
+        #Validations
+        if not isinstance(body, dict):
+            return g.auth_texts.get('ajax_error'), 400
+        if not isinstance(body.get('accounts'), list):
+            return "accounts should be a list!", 400
+
+        usernames = []
+        mails = []
+
+        # Validation for correct types and duplicates
+        for account in body.get('accounts', []):
+            if not isinstance(account.get('username'), str):
+                return g.auth_texts.get('username_invalid'), 400
+            if not isinstance(account.get('mail'), str) or not utils.valid_email(account.get('mail')):
+                return g.auth_texts.get('email_invalid'), 400
+            if not isinstance(account.get('password'), str):
```

Comment:
Could we perhaps extract this logic such that we have consistent rules about passwords? It would be easy to "forget" that we duplicated logic here!

Reply:
Agree! Was already thinking about re-writing all authentication stuff to `auth.py`, will store it in some nice functions which we call on single and multiple account creation.